### PR TITLE
chore: convert import auth config Error to AmplifyError

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes

Converts Error to AmplifyError in cases where imported auth is used but some other project resource requires different auth config. This PR doesn't change any of the validation logic, just throws AmplifyError instead of Error

#### Description of how you validated changes
Manually verified a few scenarios, existing tests pass
#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by evcodes — DISMISSED** [review]

**Comment by evcodes** [comment]
Can you please add some tests to capture the new AmplifyErrors? 
Similar to this: https://github.com/aws-amplify/amplify-cli/pull/12336/files#diff-83a6547e797c1c74f4cf975ec39193013e1c6cd02de90891a564b8e1af05841cR86-R99

**Review by akshbhu — APPROVED** [review]

**Review by sobolk — DISMISSED** [review]

**Comment by edwardfoyle** [comment]
I've added unit tests for cases where the code is reasonably testable. The other places would require refactoring / test tooling beyond the scope of this PR in order to add tests

**Review by evcodes — DISMISSED** [review]

**Review by goldbez — APPROVED** [review]

File: packages/amplify-category-interactions/src/provider-utils/awscloudformation/auth-helper.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const os = require('os');
 const CategoryName = 'interactions';
+const { AmplifyError } = require('amplify-cli-core');
```
Comment (sobolk) [inline packages/amplify-category-interactions/src/provider-utils/awscloudformation/auth-helper.js:3]:
```suggestion
const { AmplifyError } = require('@aws-amplify/amplify-cli-core');
```

make sure to pull latest dev. @evcodes was shuffling this stuff around.

File: packages/amplify-category-predictions/src/provider-utils/awscloudformation/prediction-category-walkthroughs/enable-guest-auth.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const os = require('os');
+const { AmplifyError } = require('amplify-cli-core');
```
Comment (sobolk) [inline packages/amplify-category-predictions/src/provider-utils/awscloudformation/prediction-category-walkthroughs/enable-guest-auth.js:2]:
```suggestion
const { AmplifyError } = require('@aws-amplify/amplify-cli-core');
```

**Review by sobolk — APPROVED** [review]

**Review by lazpavel — APPROVED** [review]

**Pull request merged**